tasks.py

```diff
@@ -0,0 +1,220 @@
+import os
+import sys
+import json
+import requests
+import tempfile
+import zipfile
+from shutil import rmtree, copy2, which
+from distutils.dir_util import copy_tree
+from io import BytesIO
+
+from urllib.request import urlopen #used to retrieve the proto files from github.
+from http.client import HTTPResponse
+
+from invoke import task
+from galaxy.tools import zip_folder_to_file
+
+BASE_DIR = os.path.abspath(os.path.dirname(__file__))
+PROTOC_DIR = os.path.join(BASE_DIR, "protoc")
+
+with open(os.path.join(BASE_DIR, "src", "manifest.json"), "r") as f:
+    MANIFEST = json.load(f)
+
+if sys.platform == 'win32':
+    DIST_DIR = os.environ['localappdata'] + '\\GOG.com\\Galaxy\\plugins\\installed'
+    PLATFORM = "win32"
+    
+    if which("py"):
+        PYTHON_EXE = "py -3.7"
+    else:
+        PYTHON_EXE = "python"
+
+    PROTOC_EXE = os.path.join(PROTOC_DIR, "bin", "protoc.exe")
+    PROTOC_INCLUDE_DIR = os.path.join(PROTOC_DIR, "include")
+    PROTOC_DOWNLOAD_URL = "https://github.com/protocolbuffers/protobuf/releases/download/v3.19.4/protoc-3.19.4-win32.zip"
+
+elif sys.platform == 'darwin':
+    DIST_DIR = os.path.realpath(os.path.expanduser("~/Library/Application Support/GOG.com/Galaxy/plugins/installed"))
+    PLATFORM = "macosx_10_13_x86_64"  # @see https://github.com/FriendsOfGalaxy/galaxy-integrations-updater/blob/master/scripts.py
+    PYTHON_EXE = "python"
+
+    PROTOC_EXE = os.path.join(PROTOC_DIR, "bin", "protoc")
+    PROTOC_INCLUDE_DIR = os.path.join(PROTOC_DIR, "include")
+    PROTOC_DOWNLOAD_URL = "https://github.com/protocolbuffers/protobuf/releases/download/v3.19.4/protoc-3.19.4-osx-x86_64.zip"
+
+
+@task
+def build(c, output='output', ziparchive=None):
+    if os.path.exists(output):
+        print('--> Removing {} directory'.format(output))
+        rmtree(output)
+
+    print('--> Fixing a pip issue, failing to import `BAR_TYPES` from `pip._internal.cli.progress_bars`')
+    c.run(PYTHON_EXE + ' -m pip install --upgrade pip==22.0.4')
+
+    # Firstly dependencies need to be "flattened" with pip-compile,
+    # as pip requires --no-deps if --platform is used.
+    print('--> Flattening dependencies to temporary requirements file')
+    with tempfile.NamedTemporaryFile(mode="w", delete=False) as tmp:
+        c.run(f'pip-compile requirements/app.txt --output-file=-', out_stream=tmp)
```

Comment:
This is actually a really good idea, because i just had my refactor version of the plugin break when a dependency of a dependency updated and it wasn't pinned there (they had a minimum version pinned, but not a max). But if this is something that we need to do for most plugins it's probably something we could issue out and look at for a bunch of plugins as well.

Reply:
Case by case seems fair to me. Then build action could be faster by skipping the compile step.

I think it is possible that pip-compile during build step was to deal with different compilation result depending on OS. So in the worst case scenario we will have to have 2 locks. Or one with combined OS constrains, because it is possible.

I like poetry, but after more than a year experience in using it for multiple "non-standard" setup in my work, I definitely prefer pip&pip-tools pair as a tool to cover more corner cases. It is one tool to serve one purposes and IMO it does it pretty well.

Reply:
Full disclosure, it's not my build tool, another dev added it. I'm assuming it wasn't originally for steam but they found it elsewhere and ported it. I'm of the opinion that whatever works is what's best, so i don't exactly have a preference. But we could make an `invoke` command that generates that and store it in the cache. If we choose to make a change to the requirements, then update that file with the `inv` call

Reply:
There's a mac sepcific dependency, unfortunately. This mainly deals with how we launch games. It's old code, maybe there's a better way, idk. 

This is where it's in the requirements
https://github.com/ABaumher/galaxy-integration-steam/blob/7e945d3c36dc0b6daac1fabacdf7ecc0285ceab4/requirements/app.txt#L6
This is where the dependencies are used:
https://github.com/ABaumher/galaxy-integration-steam/blob/master/src/uri_scheme_handler.py
And this is where it's called:
https://github.com/ABaumher/galaxy-integration-steam/blob/7e945d3c36dc0b6daac1fabacdf7ecc0285ceab4/src/plugin.py#L246-L253

We can also install `certifi` on macos to try to kill the annoying certificate issue on MacOS Calatina and future versions, but i haven't had much success there. 
@urwrstkn8mare would be more helpful on how to handle that, though

Reply:
thanks for chekcing, we should do it later https://github.com/GOG-Nebula/.github/issues/16